addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -231,8 +231,9 @@ def write(self, vals):
         return res
 
     def unlink(self):
-        if self.requisition_id.state not in ['draft', 'cancel', 'done']:
-            self.supplier_info_ids.unlink()
+        for requisition in self:
```

Comment:
```suggestion
        to_unlink = self.filtered(lambda r: r.requisition_id.state not in ['draft', 'cancel', 'done'])
        to_unlink.mapped('supplier_info_ids').unlink()
```

better as the unlink is done in batch